sys/src/9/pc/l.s

```diff
@@ -671,9 +671,19 @@ TEXT lcycles(SB),1,$0
 	RDTSC
 	RET
 
+// WIP: if the rdmsr fails, then return all -1
+// TODO: change type signature of rdmsr
 TEXT rdmsr(SB), $0				/* model-specific register */
 	MOVL	index+0(FP), CX
+TEXT rdmsr_doit(SB), $0
 	RDMSR
+TEXT rdmsr_ok(SB), $0
+	JMP 1f
```

Comment:
I don't know the assembler particularly - is this JMP correct?  Your label is 1, but the JMP to 1f.

Reply:
we're sending you back to school. This is a Knuth invention, in Unix assemblers for 50 years. :-)

https://stackoverflow.com/questions/27353096/1b-and-1f-in-gnu-assembly

it's incredibly handy.

Reply:
I'll go ahead and push this if tests show it does no harm